errors.go

```diff
@@ -10,28 +10,29 @@ var (
 	// ErrInf signifies the initial function value is Inf.
 	ErrInf = errors.New("optimize: initial function value is Inf")
 
-	// ErrGradInf signifies the initial function value is Inf.
-	ErrGradInf = errors.New("optimize: initial gradient is Inf")
-
-	// ErrLinesearchFailure signifies the linesearch has iterated too many
-	// times. This may occur if the gradient tolerance is set too low.
-	ErrLinesearchFailure = errors.New("linesearch: failed to converge")
-
 	// ErrNaN signifies the initial function value is NaN.
 	ErrNaN = errors.New("optimize: initial function value is NaN")
 
-	// ErrGradNaN signifies the initial function value is NaN.
-	ErrGradNaN = errors.New("optimize: initial gradient is NaN")
+	// ErrGradInf signifies that a component of the initial gradient is Inf.
+	ErrGradInf = errors.New("optimize: initial gradient is Inf")
```

Comment:
From looking through the stanard library, it looks like "optimize: initial gradient is infinite". The general rule is "Use English".